docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.install

```diff
@@ -466,3 +466,107 @@ function hs_paragraph_types_update_10013(&$sandbox) {
+    $existing_config->save();
+  }
+
+  // Update field_spotlight_style field storage.
+  $field_storage_config = $entity_type_manager->getStorage('field_storage_config');
+  $config_data = $config_directory->read('field.storage.paragraph.field_spotlight_style');
+  $existing_config = $field_storage_config->load($config_data['id']);
+  if ($existing_config) {
+    $existing_config = $field_storage_config
+      ->updateFromStorageRecord($existing_config, $config_data);
+    $existing_config->save();
+  }
+}
+
+/**
+ * Updates spotlight paragraphs with new default values.
+ */
+function hs_paragraph_types_update_10015(&$sandbox) {
+
+  $entity_type_manager = \Drupal::entityTypeManager();
+  $paragraph_storage = $entity_type_manager->getStorage('paragraph');
+
+  // Initialize the batch.
+  if (!isset($sandbox['total'])) {
+    $sandbox['ids'] = $paragraph_storage->getQuery()
+      ->accessCheck(FALSE)
+      ->condition('type', 'hs_spotlight')
+      ->execute();
+    $sandbox['total'] = count($sandbox['ids']);
+    $sandbox['current'] = 0;
+
+    if (empty($sandbox['total'])) {
+      $sandbox['#finished'] = 1;
+      return t('No spotlight paragraphs found to update.');
+    }
+  }
+
+  // Process paragraphs in batches of 25.
+  $paragraph_ids = array_slice($sandbox['ids'], $sandbox['current'], 25);
+
+  foreach ($paragraph_storage->loadMultiple($paragraph_ids) as $paragraph) {
+    $save_needed = FALSE;
+
+    // Update field_hs_spotlight_height if _none or empty.
+    if ($paragraph->hasField('field_hs_spotlight_height')) {
+      $height_value = $paragraph->get('field_hs_spotlight_height')->value;
+      if (empty($height_value) || $height_value === '_none') {
+        $paragraph->set('field_hs_spotlight_height', 'default');
+        $save_needed = TRUE;
+      }
+    }
+
+    // Update field_hs_spotlight_image_align if _none or empty.
+    if ($paragraph->hasField('field_hs_spotlight_image_align')) {
+      $align_value = $paragraph->get('field_hs_spotlight_image_align')->value;
+      if (empty($align_value) || $align_value === '_none') {
+        $paragraph->set('field_hs_spotlight_image_align', 'image-right');
+        $save_needed = TRUE;
+      }
+    }
+
+    // Update field_spotlight_style if _none or empty.
+    if ($paragraph->hasField('field_spotlight_style')) {
+      $style_value = $paragraph->get('field_spotlight_style')->value;
+      if (empty($style_value) || $style_value === '_none') {
+        $paragraph->set('field_spotlight_style', 'classic');
+        $save_needed = TRUE;
+      }
+    }
+
+    if ($save_needed) {
+      $paragraph->save();
+    }
+
+    $sandbox['current']++;
+  }
+
+  // Update progress.
+  $sandbox['#finished'] = empty($sandbox['total']) ? 1 : ($sandbox['current'] / $sandbox['total']);
+
+  if ($sandbox['#finished'] >= 1) {
+    return t('Updated @count spotlight paragraphs with new default values.', [
+      '@count' => $sandbox['total'],
```

Comment:
Only paragraphs that were saved will have been updated which may not be the total count, if you want to add a separate counter for this.